os_stub/spdm_device_secret_lib_sample/lib.c

```diff
@@ -1495,6 +1495,15 @@ uint8_t m_libspdm_bin_str0[0x11] = {
     0x64, 0x65, 0x72, 0x69, 0x76, 0x65, 0x64,
 };
 
+void libspdm_dump_hex_str_sample(const uint8_t *buffer, size_t buffer_size)
```

Comment:
Please make a separate pull request for this change.

Reply:
Agree with Steven. Rule is: each PR should only address one problem.
I am not sure why this change is relevant to shared library.

Would you please clarify: Why we cannot use old function?

Reply:
This change is required only for the shared library PR, that is why I included it here.
The regular libspdm_dump_hex_str() method is defined in multiple places, but only inside the "unit_test" folder of libspdm, so it is not available for code outside the scope of unit tests, like spdm_secret_lib_sample() is.
Regular integrations of libspdm with static libraries usually include at least one of the unit test libraries that contain an implementation of the method, so it is not an issue there.
Considering this, is it ok to keep this change as part of this PR?

Reply:
Please help me understand why os_stub/spdm_device_secret_lib_sample/lib.c is  impacted.

This is just a **sample**. Should not be used by other people, except unit test.

Do you want to create a shared lib for spdm_device_secret_lib_sample ?

Reply:
You are right, of course. SPDM_device_secret_lib_sample should not be part of the shared library, and it can be implemented in the application using the library. So there is no need for the change and I will remove it from shared library target in the next commit.